### DISCUSSION TITLE
Clarity needed on component naming conventions & the current roadmap & the documentation

### DISCUSSION THREAD

**Discussion by MarkStega** [material-components/material-web#4966] [General]
With the new roadmap I am unclear on what constitutes a 'component'.

As an example, there used to be 'button', 'floating action button', 'icon button', and 'segmented button' as separate components. Now there is only 'button' in the current release (but segmented button shows in the next release). 

When I look at 'https://material-web.dev/' I still see 'button', 'floating action button', and 'icon button' as separate components. However, I don't see 'elevation' and 'focus ring' which are listed as components in the roadmap.

Icon seems to be missing from both the component list in the roadmap as well as the documentation.

I know that this is a bit of 'moving target', but I'd like to understand what constitutes the likely expected endpoint.

The reason that I care is that we are working on our next implementation of Material.Blazor that is targeting MD3 and we'd like to be consistent with what is actually getting implemented.

**Comment by asyncliz** [comment]
Typically a "component" is defined by design on https://m3.material.io/components.

However, as a web library we provide additional components that aren't explicitly defined in the spec but are critical to using Material on the web, such as select, icon, ripple, and elevation.

>  Now there is only 'button' in the current release

Button, FAB, and icon button are all still part of the 1.0 release. Segmented buttons have been planned for a future update.

**Comment by MarkStega** [comment]
Liz,

Thanks for the reply; The confusion that I had was that "Component" is really referring to two different concepts. Just to make certain that I am not mangling your answer:

At the top level are m3.material.io components such as buttons, progress indicators, and text fields.

Underlying those components are web components. A material component may be implemented with one or more web components. As a concrete example, the Material "progress indicators" has two underlying web components "md-circular-progress" and "md-linear-progress".

In a sense, then the components such as select and icon can be considered m3.material.io "Plus" components required for a good web implementation.
